lib/guardian/topic_guardian.rb

```diff
@@ -70,8 +70,7 @@ def can_move_topic_to_category?(category)
           Category.find(category || SiteSetting.uncategorized_category_id)
         end
       )
-
-    is_staff? || (can_create_topic_on_category?(category) && !category.require_topic_approval?)
+    is_staff? || can_create_topic_on_category?(category) && !topics_need_approval?(category)
```

Comment:
The real change is here. Because `require_topic_approval?` shouldn't entirely block a category moderator from moving a post.